lib/inets/src/http_lib/http_uri.erl

```diff
@@ -302,19 +313,19 @@ path("") ->
 path(Path) ->
     Path.
 
-uri_encode(Char, Reserved) ->
-    case sets:is_element(Char, Reserved) of
-	true ->
-	    [ $% | http_util:integer_to_hexlist(Char)];
+uri_encode(Char) ->
+    case is_unreserved(Char) of
 	false ->
+	    [ $% | http_util:integer_to_hexlist(Char)];
```

Comment:
I see why you wrote it this way, but it leads to a double-negation (is_unreserved() = false) which makes it awkward to read.

Reply:
I tried to minimize the diff, but it also matches the documentation: if not in the unreserved set, escape it, otherwise leave it as is.

Reply:
Also, in the URI terminology, "unreserved" is a different thing from "not reserved".